step-capstone/src/scripts/Suggestion.js

```diff
@@ -0,0 +1,77 @@
+import queryPlaces from './PlacesQuery';
+import categories from './Categories';
+
+const query = (coordinates, service) => {
+    // return results: a map with key : place_id, value: PlaceObject
+    let places = queryPlaces(coordinates, radius, service, ["tourist_attraction", "natural_feature"])
+    places.then(results => {
+        return results;
+    })
+}
+
+const countCat = (placeObject, userCat) => {
+    // Count the number of categories a place fits into
+    let cats = 0;
+    userCat.forEach(catString => {
+        placeObject.place.types.forEach(type => {
+            if (categories.get(catString).has(type)) {
+                cats += 1;
+                break;
+            }
+        })
+    
+    })
+    return cats;
+}
+
+const getScore = (placeCat, userCat, prominence, placePrice, userBudget, rating) => {
+    // return the score for a PlaceObject
+    var catScore = 0;
+    if (cat !== 0) {
+        catScore = 60 + 40/userCat * placeCat;
```

Comment:
If I understand this part correctly, we expect that userCat > placeCat for all scenarios so 40 is never multiplied to a value bigger than 1. A few suggestions here:

1. Add checks to ensure that the conditions I've mentioned is true before calculating catScore
2. Express catScore to be 60 + 40 * (placeCat / userCat). This seems to make it more intuitive to understand

Reply:
We changed some of the naming here which was not very good. userCat is now renamed preferenceCategories - it is the total number of user preference categories. placeCat, now renamed matchingCategories, is the number of those preferences that match the categories in placeObject. So preferenceCategories is always smaller than matching Categories.